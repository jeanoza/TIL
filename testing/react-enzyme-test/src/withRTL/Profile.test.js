import Profile from "../counters/Profile";
import { render } from "@testing-library/react";

describe("<Profile/>", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="jeanoza" name="Kyubong" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(<Profile username="jeanoza" name="Kyubong" />);
    utils.getByText("jeanoza");
    utils.getByText("(Kyubong)");
    utils.getByText(/kyu/i); // passe avec regex
  });
});
