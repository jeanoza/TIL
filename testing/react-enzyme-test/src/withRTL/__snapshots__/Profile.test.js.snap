// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<div>
  <div>
    <b>
      jeanoza
    </b>
    Â 
    <span>
      (
      Kyubong
      )
    </span>
  </div>
</div>
`;
